test/foundry/unit/TokenDistributor.t.sol

```diff
@@ -0,0 +1,1446 @@
+        goForward(1 weeks);
+        vm.expectRevert(
+            abi.encodeWithSelector(ITokenDistributor.CannotClaim0Fees.selector)
+        );
+        tokenDistributor.claimEpoch(address(user1), 0);
+    }
+
+    /// @notice claimEpoch fail - nonstaker tries to claim
+    /// (cannot claim 0 fees)
+    function testClaimNotStaker() public {
+        //setup
+        kwenta.transfer(address(user1), 1);
+        vm.startPrank(address(user1));
+        kwenta.approve(address(stakingRewardsV2), 1);
+        stakingRewardsV2.stake(1);
+        vm.stopPrank();
+        goForward(1 weeks);
+        kwenta.transfer(address(tokenDistributor), 10);
+        goForward(1 weeks);
+
+        tokenDistributor.claimEpoch(address(user1), 1);
+        vm.expectRevert(
+            abi.encodeWithSelector(ITokenDistributor.CannotClaim0Fees.selector)
+        );
+        tokenDistributor.claimEpoch(address(user2), 1);
+    }
+
+    /// @notice claimEpoch happy case with a person who
+    /// was previously staked but is now unstaked and trying to
+    /// claim their fees
+    function testClaimPreviouslyStaked() public {
+        //setup
+        kwenta.transfer(address(user1), 1);
+        vm.startPrank(address(user1));
+        kwenta.approve(address(stakingRewardsV2), 1);
+        stakingRewardsV2.stake(1);
+        vm.stopPrank();
+        goForward(1 weeks);
+        kwenta.transfer(address(tokenDistributor), 10);
+        goForward(1 weeks);
+
+        vm.prank(address(user1));
+        stakingRewardsV2.unstake(1);
+        goForward(2 weeks);
+        vm.expectEmit(true, true, false, true);
+        emit VestingEntryCreated(address(user1), 2, 31449600, 1, 90);
+        tokenDistributor.claimEpoch(address(user1), 1);
+    }
+
+    /// @notice testCalculateEpochFees happy case
+    function testCalculateEpochFees() public {
+        kwenta.transfer(address(user1), 1);
+        kwenta.transfer(address(user2), 2);
+        vm.startPrank(address(user1));
+        kwenta.approve(address(stakingRewardsV2), 1);
+        stakingRewardsV2.stake(1);
+        vm.stopPrank();
+        vm.startPrank(address(user2));
+        kwenta.approve(address(stakingRewardsV2), 2);
+        stakingRewardsV2.stake(2);
+        vm.stopPrank();
+
+        goForward(1 weeks);
+        /// @dev forward half a week so it puts fees in epoch 1
+        goForward(304801);
+        kwenta.transfer(address(tokenDistributor), 1000);
+        tokenDistributor.checkpointToken();
+        goForward(1 weeks);
+
+        uint256 sum = tokenDistributor.calculateEpochFees(user1, 1);
+        assertEq(sum, 111);
+    }
+
+    /// @notice fuzz calculateEpochFees, especially when checkpoints are across weeks
+    function testFuzzCalculateEpochFees(uint256 amount) public {
+        /// @dev make sure its less than this contract
+        /// holds and greater than 10 so the result isn't
+        /// 0 after dividing
+        vm.assume(amount < 100_000 ether);
+        vm.assume(amount > 10);
```

Comment:
Does the contract have issues if the divisible amount of tokens is below 10? Meaning are those < 10 KWENTAs lost forever?

Reply:
In situations where there is < 10 KWENTA and theres a large amount of staking and/or lots of stakers (compared to tokens), yes those KWENTA can't be claimed and are lost forever.